src/org/thoughtcrime/securesms/util/DynamicLanguage.java

```diff
@@ -32,7 +32,7 @@ public void onCreate(Activity activity) {
   }
 
   public void onResume(Activity activity) {
-    if (!currentLocale.equals(getSelectedLocale(activity))) {
+    if (!getSelectedLocale(activity).equals(currentLocale)) {
```

Comment:
currentLocale could be null and you can't call equals() on null, while the other value is always a Locale object, while testing it happened to me and got exception, but was also due to other bugs while doing the modifications so maybe it would work but it doesn't hurt